src/test/scala/sangria/marshalling/EnumInputTypeSpec.scala

```diff
@@ -0,0 +1,105 @@
+package sangria.marshalling
+
+import java.util.concurrent.Executor
+
+import org.scalatest.{Matchers, WordSpec}
+import sangria.marshalling.sprayJson.SprayJsonInputUnmarshaller
+import sangria.parser.QueryParser
+import sangria.schema.{Argument, EnumType, EnumValue, Field, InputField, InputObjectType, ObjectType}
+import spray.json.{JsObject, JsString, JsValue, JsonReader}
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext}
+import scala.util.Success
+
+class EnumInputTypeSpec extends WordSpec with Matchers {
+  implicit val inputUnmarshaller: SprayJsonInputUnmarshaller.type = SprayJsonInputUnmarshaller
+
+  implicit def sprayJsonFromInput[T <: JsValue]: FromInput[T] = sprayJson.sprayJsonFromInput[T]
+
+  implicit def sprayJsonReaderFromInput[T: JsonReader]: FromInput[T] = sprayJson.sprayJsonReaderFromInput[T]
+
+  private val enumWithCustomValueType = EnumType[String](
+    "CustomEnum",
+    Some("Some custom enum values"),
+    List(
+      EnumValue("TOP", value = "TOP_VALUE"),
+      EnumValue("OTHER", value = "OTHER_VALUE")
+    )
+  )
+
+  private val complexInputType = InputObjectType[JsObject]("customInput", fields = List(InputField("enumValue", enumWithCustomValueType)))
+  private val complexArgument  = Argument("complexArgument", complexInputType)
+  private val enumArgument     = Argument("simpleArgument", enumWithCustomValueType)
+
+  private val valueFromComplexField: Field[Unit, JsValue] = {
+    Field(
+      "valueFromComplex",
+      sangria.schema.StringType,
+      arguments = List(complexArgument),
+      resolve = { ctx =>
+        ctx.arg(complexArgument).getFields("enumValue").toString()
```

Comment:
@OlegIlyenko There is a small mistake here, we should get the head instead of picking the string representation of the `Seq` but anyways the test still fails

Reply:
👍